code/http/url/url.php

```diff
@@ -356,20 +343,28 @@ public function getPort()
      */
     public function setPort($port)
     {
-        $this->port = $port;
+        $this->port = (int) rawurldecode($port);
         return $this;
     }
 
     /**
      * Returns the path portion as a string or array
      *
-     * @param     boolean $toArray If TRUE return an array. Default FALSE
+     * This method will encode the path to conform to RFC 3986 returned as string.
+     * @link https://tools.ietf.org/html/rfc3986
+     *
+     * @param   boolean $toArray If TRUE return an array. Default FALSE
      * @return  string|array The path string; e.g., `path/to/site`.
```

Comment:
@johanjanssens need to update the phpdoc to /path/to/site as there will always be a leading slash now.